### PR TITLE
Use SA config when creating clients

### PR DESCRIPTION
**Pull request by ironcladlou**
Build client configs using SA information (e.g. service accounts host,
mounted service account auth files).

This fixes possible connectivity issues in multi-master setups.

### CODE REVIEW

**Comment by ironcladlou** [comment]
cc @liggitt @detiber

File: pkg/cmd/infra/deployer/deployer.go [hunk 1]
```diff
@@ -45,20 +45,21 @@ func NewCommandDeployer(name string) *cobra.Command {
 		Short: "Run the deployer",
 		Long:  deployerLong,
 		Run: func(c *cobra.Command, args []string) {
-			_, kClient, err := cfg.Config.Clients()
```
Comment (liggitt) [inline pkg/cmd/infra/deployer/deployer.go]: is `cfg.Config` used any more?

**Comment by liggitt** [comment]
I assume you're not removing the envvars set by the deployer pod controller now to support old deployer images for a bit... can you open a follow up pull or issue to remove that?

File: pkg/cmd/infra/deployer/deployer.go (hunk 1 above)
Comment (ironcladlou) [inline pkg/cmd/infra/deployer/deployer.go]: Nope, removed.

**Comment by liggitt** [comment]
LGTM, [test]

**Comment by ironcladlou** [comment]
>I assume you're not removing the envvars set by the deployer pod controller now to support old deployer images for a bit... can you open a follow up pull or issue to remove that?

Created https://github.com/openshift/origin/issues/7199.

**Comment by liggitt** [comment]
```
--- /data/src/github.com/openshift/origin/contrib/completions/bash/openshift	2016-02-10 14:11:19.545074782 -0500
+++ /data/src/github.com/openshift/origin/_output/verify-generated-completions//contrib/completions/bash/openshift	2016-02-10 14:18:57.221417194 -0500
@@ -17898,15 +17898,8 @@ _openshift_infra_deploy()
     flags_with_completion=()
     flags_completion=()
 
-    flags+=("--certificate-authority=")
-    flags+=("--client-certificate=")
-    flags+=("--client-key=")
     flags+=("--deployment=")
-    flags+=("--insecure-skip-tls-verify")
-    flags+=("--kubernetes=")
-    flags+=("--master=")
     flags+=("--namespace=")
-    flags+=("--token=")
     flags+=("--google-json-key=")
     flags+=("--log-flush-frequency=")
```

**Comment by liggitt** [comment]
glad we can still pass `--google-json-key` to our deployer...

**Comment by detiber** [comment]
@liggitt wouldn't you know that is the one I use the most!

**Comment by ironcladlou** [comment]
[test] again, looks flaky

**Comment by liggitt** [comment]
LGTM, [merge]

**Pull request merged**